### PR TITLE
3.0 Travis CI integration & build system overhaul [Ready for review!]

### PR DESCRIPTION
**Pull request by bauerm97**
**Description of the Pull Request (PR):**

- Set up travis CI for 3.0 tree, we can finally have testing!
- Reconfigure entire build system. Compiling project is now very easy:
```
./mconfig -b ./buildtree
make -C ./buildtree
sudo make -C ./buildtree install
```
- Fix as many hardcoded paths as I could find. Now properly using buildtime configurables which are generated by `pkg/buildcfg/confgen` into `pkg/buildcfg`.
- A bit of cleanup to mconfig and our Makefile.stub
- This will also serve as a backbone for the reorganization of the repository being done by @yhcote, so we are working in a cleaner tree.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+  - dep ensure -vendor-only
+
+install:
+  - ./mconfig -b ./buildtree -p /usr/local
+  - make -j `nproc 2>/dev/null || echo 1` -C ./buildtree
+  - sudo make -C ./buildtree install
 
 script:
-  - .travis/script
+  - go test ./tests/
```
Comment (tri-adam) [inline .travis.yml:32]: I'd strongly encourage adding `go fmt`, `go vet`, and `golint` in the CI (and perhaps during the `make`.)

`go fmt ./...` is clean in your branch and could easily be added as part of this PR.

`go vet` and `golint` are both finding issues and might be best addressed separately (I have added comments for new `go vet` issues introduced by this PR.)

File: internal/pkg/cli/build.go [hunk 2]
```diff
@@ -59,26 +59,26 @@ var buildCmd = &cobra.Command{
 			// URI passed as arg[1]
 			def, err = build.NewDefinitionFromURI(args[1])
 			if err != nil {
-				fmt.Println("Error: ", err)
-				return
+				fmt.Println("Unable to parse URI %s: ", args[1], err)
```
Comment (tri-adam) [inline internal/pkg/cli/build.go:62]: From `go vet`:

```
internal/pkg/cli/build.go:62: possible formatting directive in Println call
```

File: internal/pkg/cli/build.go [hunk 3]
```diff
@@ -59,26 +59,26 @@ var buildCmd = &cobra.Command{
...
+				os.Exit(1)
 			}
 		} else if !ok && err == nil {
 			// Non-URI passed as arg[1]
 			defFile, err := os.Open(args[1])
 			if err != nil {
-				fmt.Println("Error: ", err)
-				return
+				fmt.Println("Unable to open file %s: ", args[1], err)
```
Comment (tri-adam) [inline internal/pkg/cli/build.go:69]: From `go vet`:

```
internal/pkg/cli/build.go:69: possible formatting directive in Println call
```

File: internal/pkg/cli/build.go [hunk 4]
```diff
@@ -59,26 +59,26 @@ var buildCmd = &cobra.Command{
...
-				fmt.Println("Error: ", err)
-				return
+				fmt.Println("Failed to parse definition file %s: ", args[1], err)
+				os.Exit(1)
 			}
 		} else {
 			// Error
-			fmt.Println("Error: ", err)
-			return
+			fmt.Println("Unable to parse %s: ", args[1], err)
```
Comment (tri-adam) [inline internal/pkg/cli/build.go:80]: From `go vet`:

```
internal/pkg/cli/build.go:80: possible formatting directive in Println call
```

File: internal/pkg/cli/build.go [hunk 5]
```diff
@@ -59,26 +59,26 @@ var buildCmd = &cobra.Command{
...
-				return
+				fmt.Println("Unable to open file %s: ", args[1], err)
+				os.Exit(1)
 			}
 
 			def, err = build.ParseDefinitionFile(defFile)
 			if err != nil {
-				fmt.Println("Error: ", err)
-				return
+				fmt.Println("Failed to parse definition file %s: ", args[1], err)
```
Comment (tri-adam) [inline internal/pkg/cli/build.go:75]: From `go vet`:

```
internal/pkg/cli/build.go:75: possible formatting directive in Println call
```

**Review by tri-adam — CHANGES_REQUESTED** [review]
In general, looks great and a huge improvement. Minor suggestions for improvement only!

File: internal/pkg/cli/build.go (hunk 2 above)
Comment (ArangoGutierrez) [reply internal/pkg/cli/build.go:62]:
```go
fmt.Printf("Unable to parse URI %s:  %v", args[1], err)
```

**Pull request merged**